### ISSUE TITLE
Can't run pm2 on windows

### ISSUE THREAD

**Issue by StarpTech** [Unitech/PM2#1251]
````
fs.js:747
  return binding.mkdir(pathModule._makeLong(path),
                 ^
Error: ENOENT, no such file or directory 'E:\Users\dustin\.pm2'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:747:18)
    at Object.CLI.pm2Init (C:\Users\dustin\AppData\Roaming\npm\node_modules\pm2\lib\CLI.js:37:8)
    at Object.<anonymous> (C:\Users\dustin\AppData\Roaming\npm\node_modules\pm2\bin\pm2:20:5)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
```

Pm2 want to have access of an invalid user directory `E:\Users\dustin\.pm2`

NodeJs: v0.12.2
PM2: 0.12.13
Windows: 8.1

**Comment by StarpTech** [comment]
I got it by setting the `$env:PM2_HOME` manually.

**Comment by Shershnev-AV** [comment]
I have the same error.
I tried to install pm2 with -g and without, as admin and no, always this error

nodejs: 0.12.2
pm2: 0.12.14
windows: 10 Pro Insider Preview (build 10074)

**Comment by jshkurti** [comment]
I'm going to get a look at this

**Comment by efkan** [comment]
StarpTech  +1

**Comment by rainabba** [comment]
Given the number of issues and the general nature of this one, I'm going to piggyback it for now. I installed with -g. --version reports 0.12.16. Node is 0.10.36. It seems that I can start an app and it runs, but if I do ANYTHING to interact with pm2 afterwards such as list or monit, both my app process and the pm2 process die silently. I don't think my app is doing anything out of the norm to contribute and runs fine in dev and production using forever (was reading an article touting PM2 as an improvement so I'm checking it out).

**Comment by StarpTech** [comment]
Status?

**Comment by marklagendijk** [comment]
I have also encountered this issue. It looks like it only occurs if you execute from a path outside of `C:\` (which is where the user folder it is looking for is located).

**Comment by StarpTech** [comment]
@marklagendijk thanks!

**Comment by rainabba** [comment]
`TypeError: Cannot read property 'replace' of undefined node_modules\pm2\node_modules\pidusage\lib\stats.js:136:26` stdout is undefined which makes sense since on stats.js:135, stdout is an empty string.

If I put a check in place there and setup the stats object to have null values for the 3 children and try again, then I get an error at 152 raised by async `Error: Callback was already called.`

Was this tested on Windows? If so, what version?

I'm willing to provide direct access with TeamViewer if someone wants to investigate/test further on my system. Really looking forward to being able to use PM2.

*tl;dr*

I've updated to Node 12.0, reinstalled pm2, did `pm2 update` and see the same behavior with a completely new app. I `pm2 start index.js` and see my app only. Last try I then did `pm2 show index` (index.js) and got a result, but a 2nd try moments later of `pm2 show index` and Here's what I got:

```
[PM2] Spawning PM2 daemon
[PM2] PM2 Successfully daemonized
[PM2][WARN] index doesn't exist
```

I see an instance of Node.exe running still so I'm guessing that PM2 is still running, but without even accessing my app (so I know I didn't crash it) which runs fine on its own and with forever, I'm guessing that its thread is no longer there.

I also tried setting PM2_HOME and when I start my app, I get a conf.js and module_conf.json file in that folder. I then tried `pm2 start index.js` again and it wouldn't even start. I then killed that node.exe instance I saw and tried again. This time I got a logs and pids folder and pm2.og and pm2.pid files in that folder, but when I tried `pm2 list` it seemed to hang up and when I tried `pm2 show index` I got the same result as above.

Now that I have a log file, I'm seeing what's going on:

`TypeError: Cannot read property 'replace' of undefined node_modules\pm2\node_modules\pidusage\lib\stats.js:136:26` stdout is undefined which makes sense since on stats.js:135, stdout is an empty string.

If I put a check in place there and setup the stats object to have null values for the 3 children and try again, then I get an error at 152 raised by async `Error: Callback was already called.`

Was this tested on Windows? If so, what version?

I'm willing to provide direct access with TeamViewer if someone wants to investigate/test further on my system. Really looking forward to being able to use PM2.

**Comment by soyuka** [comment]
Can you try the development version `npm i -g Unitech/PM2#development`?

**Comment by psulek** [comment]
npm i -g Unitech/PM2#development does not work for me.
node: v0.12.4
windows server 2012 R2 x64

But when i use option ` --no-daemon` cluster is working!

**Comment by rainabba** [comment]
Neither the development version nor --no-daemon make PM2 usable. I do see different behavior with --no-daemon; it locks up my console, but then apparently the app does instantly where it otherwise runs until I try to use list or show commands.

**Comment by amatzko** [comment]
FYI I believe this is because the environmental variable is set to the pm2 file, not the pm2 folder like it should be.
